docs/src/main/asciidoc/kafka.adoc

```diff
@@ -321,7 +323,7 @@ This strategy should not be used in high load environment, as offset commit is e
 
 - `ignore` performs no commit. This strategy is the default strategy when the consumer is explicitly configured with `enable.auto.commit` to true.
 It delegates the offset commit to the underlying Kafka client.
-This strategy provides at-least-once delivery if the channel processes the message without performing any asynchronous operations and when `enable.auto.commit` is set to true.
+When `enable.auto.commit` is `true` this strategy **DOES NOT** guarantee at-least-once delivery.
```

Comment:
We already mention `enable.auto.commit` in previous sentence, no need to repeat it I guess?

```suggestion
This strategy **DOES NOT** guarantee at-least-once delivery.
```

Also, the following sentence "However, if the processing failed between two commits, messages received after the commit and before the failure will be re-processed" suggests that at-least-once kinda sorta _does_ happen?

Reply:
They describe somewhat different things.
Let's consider the following for the reprocessing scenario:
- last committed offset: 30, max queue size: 100, current queue size: 0
- poll 10 records, consumer position: 40, current queue size: 10
- processed records: 4, current queue size: 6
- FAIL before the subsequent poll (and commit), the processing takes over from offset 30. Records between 30-34 will be re-processed.

And now for the scenario in which at-least-once is not covered:
- last committed offset: 30, max queue size: 100, current queue size: 0
- poll 10 records, consumer position: 40, current queue size: 10
- processed records: 4, current queue size: 6
- poll 20 records, current position 60, auto-commit offset 40, current queue size: 26
At this point, auto-commit committed offset 40 whereas the latest processed offset is 34. If a failure or a rebalance happens, records between 34-40 will be missed.

Reply:
BTW one can choose IGNORE commit strategy with `enable.auto.commit=false`. There will be no offset commits.

Reply:
Ah OK, gotcha! This is pretty important, can we maybe expand on it? Once sentence doesn't quite does it justice :-) Something like:

> This strategy **DOES NOT** guarantee at-least-once delivery.
> SmallRye Reactive Messaging processes records asynchronously, and so offsets may be committed for records that have been polled but not yet processed.
> In case of a failure, only records that were not yet committed will be re-processed.

Reply:
👍